Add the let* monadic notation

lthms opened the pull request:
Closes #95 

Inspired by the notation introduced by OCaml 4.08, we propose to
introduce a new notation following the same principle.

```coq
    let* x := p in q
```

is strictly equivalent to

```coq
    x <- p;; q
```

lthms commented:
Sorry, I misunderstood your comment on the issue.

I was not sure how to proceed, so I propose two approach:

- Either always load `_ <- _ ;; _`, and load `let* _ := _ in _` if desired (second commit)
- Or load either one or the other (the third commit)

I personally prefer the latter, and it does not break the previous interface afaict. What do you think?

liyishuai commented:
Thanks! The latter with `MonadBaseNotation` looks better. I'll try to figure out how to make CI happy.

lthms commented:
The failure with the CI was due to a commit I had added which was breaking old Coq version (declaring the scope). I have removed it and now it appears that it works fine.

I should probably squash the two commits together if you prefer the approach with the base module (I do too! (: ). I think it would probably be wise to add some kind of test case too, right?

liyishuai commented:
Yes we could add some `Notation.v` under `/examples`.

lthms commented:
I think we are good to go now!
